File: src/main/java/arc/arc/listeners/SpawnerListener.java
package arc.arc.listeners;

import arc.arc.hooks.ArcModule;
import arc.arc.util.CooldownManager;
import arc.arc.util.TextUtil;
import net.kyori.adventure.text.Component;
import org.bukkit.Material;
import org.bukkit.NamespacedKey;
import org.bukkit.block.TileState;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.block.BlockBreakEvent;
import org.bukkit.persistence.PersistentDataContainer;

import java.util.Set;

public class SpawnerListener implements ArcModule, Listener {

    @EventHandler
    public void spawnerBreak(BlockBreakEvent event){
        if(event.getBlock().getType() != Material.SPAWNER) return;
        if(CooldownManager.cooldown(event.getPlayer().getUniqueId(), "spawner_break") > 0) return;
        if(event.getBlock().getState() instanceof TileState tileState){
            PersistentDataContainer container = tileState.getPersistentDataContainer();
            Set<NamespacedKey> keySet = container.getKeys();
            boolean res = false;
            for(NamespacedKey key : keySet){
                if(key.getNamespace().startsWith("oh_the_dungeons")){
                    res = true;
                    break;
                }
            }
            if(!res) return;
            Player player = event.getPlayer();
            player.sendActionBar(TextUtil.strip(
                    Component.text("В этом данже спавнеры не добываются!")
            ));
            CooldownManager.addCooldown(event.getPlayer().getUniqueId(), "spawner_break", 1200);
        }
    }
}

File: src/main/kotlin/ru/arc/listeners/SpawnerListener.kt
package ru.arc.listeners

import net.kyori.adventure.text.Component
import org.bukkit.Material
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.block.BlockBreakEvent
import ru.arc.util.CooldownManager
import ru.arc.util.TextUtil

class SpawnerListener : Listener {

    @EventHandler
    fun spawnerBreak(event: BlockBreakEvent) {
        if (event.block.type != Material.SPAWNER) return
        if (CooldownManager.cooldown(event.player.uniqueId, "spawner_break") > 0) return
        val tileState = event.block.state as? org.bukkit.block.TileState ?: return
        val hasOhDungeon = tileState.persistentDataContainer.keys.any {
            it.namespace.startsWith("oh_the_dungeons")
        }
        if (!hasOhDungeon) return
        val msg = TextUtil.strip(Component.text("В этом данже спавнеры не добываются!")) ?: return
        event.player.sendActionBar(msg)
        CooldownManager.addCooldown(event.player.uniqueId, "spawner_break", 1200)
    }
}
